shardmanager.go

```diff
@@ -84,19 +86,51 @@ func NewShardManager(
 
 // General
 
-func (n *ShardManager) connectShardNodes(ctx context.Context, shardID ShardIDType) error {
+// connectShardNodes connects to new shard peers until
+// we have at least `numShardPeerToConnect` number of shard peers.
+// Note that if we don't have the shard peer's listening address,
+// a DHT query will be made which will change the network topology.
+func (n *ShardManager) connectShardNodes(ctx context.Context, numShardPeerToConnect int, shardID ShardIDType) error {
 	// Add span for connectShardNodes of ShardManager
 	spanctx := logger.Start(ctx, "ShardManager.connectShardNodes")
 	defer logger.Finish(spanctx)
 	logger.SetTag(spanctx, "shard", shardID)
 
-	pinfos, err := n.discovery.FindPeers(spanctx, strconv.FormatInt(shardID, 10))
+	//Get shard peers that we already connected to
+	connectedPIDs := n.pubsubService.ListPeers(getCollationsTopic(shardID))
+	mapConnectedPIDs := make(map[peer.ID]bool)
+	for _, p := range connectedPIDs {
+		mapConnectedPIDs[p] = true
```

Comment:
If using `struct{}` as the type in map, the value assigned should be the instance in type `struct{}`
```suggestion
		mapConnectedPIDs[p] = struct{}{}
```